product_supplierinfo_for_customer_invoice/models/account_invoice.py

```diff
@@ -2,31 +2,28 @@
 #     (<http://www.agilebg.com>)
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
-from odoo import api, fields, models
+from odoo import fields, models
 
 
 class AccountInvoiceLine(models.Model):
-    _inherit = "account.invoice.line"
+    _inherit = "account.move.line"
 
-    @api.multi
-    @api.depends("product_id")
-    def _get_product_customer_code(self):
+    def _compute_product_customer_code(self):
         product_customerinfo_obj = self.env["product.customerinfo"]
-        for line in self.filtered(lambda il: il.invoice_id.partner_id.customer):
-            product = line.product_id
-            code_id = product_customerinfo_obj.search(
-                [
-                    ("product_tmpl_id", "=", product.product_tmpl_id.id),
-                    ("name", "=", line.invoice_id.partner_id.id),
-                ],
-                limit=1,
-            )
+        for line in self:
+            code_id = product_customerinfo_obj.browse()
+            if line.move_id.move_type in ["out_invoice", "out_refund"]:
```

Comment:
You can use [this](https://github.com/odoo/odoo/blob/14.0/addons/account/models/account_move.py#L2043) here ;)

Reply:
done, thanks